kroxylicious-additional-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/BufferPool.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.filter.encryption;
+
+import java.nio.ByteBuffer;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.atomic.AtomicReferenceArray;
+
+interface BufferPool {
+
+    ByteBuffer acquire(int size);
+
+    void release(ByteBuffer buffer);
+
+    static BufferPool allocating() {
+        return new BufferPool() {
+            @Override
+            public ByteBuffer acquire(int size) {
+                return ByteBuffer.allocate(size);
+            }
+
+            @Override
+            public void release(ByteBuffer buffer) {
+
+            }
+        };
+    }
+
+    static BufferPool directAllocating() {
+        return new BufferPool() {
```

Comment:
Qpid taught us that there's lots of nasty hallowe'en surprises if you rely on the JVM to dellocate your DBBs.  IIRC the JVM keeps a queue of DBBs to release which it cleans periodically.  We saw OOM (direct) issues until we introduced a caching strategy.

Reply:
This is fair. I don't think there are any perf benefits to be had by using direct BB anyway for how the returned BB are used. So I'll remove it.